lib/traject/macros/iiif.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module Macros
+  # Macros for extracting Stanford Specific MODS values from Nokogiri documents
+  module IIIF
+    def grab_iiif_manifest(manifest)
+      resp = Faraday.get URI.parse(manifest)
+      ::JSON.parse(resp.body) if resp.status == 200
+    end
+
+    def iiif_thumbnail_id(iiif_json)
+      iiif_json.dig('thumbnail', '@id')
+    end
+
+    def iiif_thumbnail_service_id(iiif_json)
+      iiif_json.dig('thumbnail', 'service', '@id')
+    end
+
+    def iiif_thumbnail_service_protocol(iiif_json)
+      iiif_json.dig('thumbnail', 'service', 'profile')
+    end
+
+    def iiif_thumbnail_service_conforms_to(iiif_json)
+      iiif_service_conforms_to(iiif_json.dig('thumbnail', 'service', 'profile'))
+    end
+
+    def iiif_sequence_id(iiif_json)
+      rep_iiif_resource(iiif_json).dig('@id') if rep_iiif_resource(iiif_json).present?
+    end
+
+    def iiif_sequence_service_id(iiif_json)
+      rep_iiif_resource(iiif_json).dig('service', '@id') if rep_iiif_resource(iiif_json).present?
+    end
+
+    def iiif_sequence_service_protocol(iiif_json)
+      rep_iiif_resource(iiif_json).dig('service', 'profile') if rep_iiif_resource(iiif_json).present?
+    end
+
+    def iiif_sequence_service_conforms_to(iiif_json)
+      iiif_service_conforms_to(rep_iiif_resource(iiif_json).dig('service', 'profile')) if rep_iiif_resource(iiif_json).present?
+    end
+
+    def rep_iiif_resource(manifest_json)
```

Comment:
If you did something like:

```
manifest_json['sequences'].first['canvases'].first['images'].first['resource'] || {}
```

you wouldn't need to check `if rep_iiif_resource(iiif_json).present?` above, which might be nice.